### PR TITLE
🔥 🚀 Fixing torpedoes

### PR DESCRIPTION
**Pull request by illmatah**
These commits are supposed to fix the Torpedoes, the code for which were full of bugs.

### CODE REVIEW

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 1]
```diff
@@ -13,6 +13,7 @@ public class TorpedoStore {
   private double FAILURE_RATE = 0.0; //NOSONAR
 
   private int torpedoCount = 0;
+  Random r = new Random();
```
Comment (imrebalint17) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:16]: Nice

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 2]
```diff
@@ -30,18 +31,16 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+       throw new IllegalArgumentException("numberOfTorpedos");
```
Comment (imrebalint17) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:34]: Awesome

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 3]
```diff
@@ -30,18 +31,16 @@ public TorpedoStore(int numberOfTorpedos){
...
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
-    double r = generator.nextDouble();
 
-    if (r >= FAILURE_RATE) {
+    if (r.nextDouble() >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
```
Comment (imrebalint17) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:43]: Great

File: src/test/java/hu/bme/mit/spaceship/GT4500Test.java [hunk 4]
```diff
@@ -1,11 +1,10 @@
-//This is a very angry comment to show how dissatisfied I am with the code here
 package hu.bme.mit.spaceship;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
-import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.*;  //This is an ambivalent edit
```
Comment (imrebalint17) [inline src/test/java/hu/bme/mit/spaceship/GT4500Test.java:8]: Yes it is

**Review by imrebalint17 — CHANGES_REQUESTED** [review]
It's perfect

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java (hunk 1 above)
Comment (illmatah) [reply src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:16]: thank you

File: src/test/java/hu/bme/mit/spaceship/GT4500Test.java (hunk 4 above)
Comment (illmatah) [reply src/test/java/hu/bme/mit/spaceship/GT4500Test.java:8]: sooo ambivalent amirite

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java (hunk 2 above)
Comment (illmatah) [reply src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:34]: 🐶

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java (hunk 3 above)
Comment (illmatah) [reply src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:43]: 🚒

**Review by imrebalint17 — APPROVED** [review]

**Pull request merged**